### ISSUE TITLE
Fine-tuning for higher resolutions

### ISSUE THREAD

**Issue by gemasphi** [AliaksandrSiarohin/first-order-model#390]
Hey, I was thinking of fine-tuning the already pretrained 256x256 model for higher resolutions. I have a few questions:

1. Is this even a viable option? I was thinking of maybe fine-tuning it to 720x720, however, I'm afraid that the resolution (64x64) used for the keypoint detector and dense-motion will be too low for this input resolution - which might decrease the animation quality.
2. If I wanted to fine-tune it to input images with a ratio different from 1:1, I would only have to edit the AntiAliasInterpolation2d class, and make it use a non-square kernel and have a different scale for each axis, correct? Or is there anything else that I should take into consideration besides that class?  

Thanks!

**Comment by AliaksandrSiarohin** [comment]
1. True, this could be. But with larger resolution  keypoints may become unstable. You can discil output from low resolution to high resolution, E.g. add losses that make these outputs similar.
2. Changing that class is not needed I guess. If for example you want 512x256 then it will just make it 128x64 which is also fine. I did not checked with non 1x1 ratio, but I tried to write it in such a way that it could be applied to any ratio. But of course there maybe some bugs, like inverted height and width.

**Comment by gemasphi** [comment]
1. I didn't quite understand what you mean by distil output from low res to high res.
2. I wouldn't need to change if I were training from zero, but to reuse the already trained weights, I need to guarantee that the input to the keypoint detector is 64x64.

**Comment by AliaksandrSiarohin** [comment]
1. When training on high res, add a loss |kp_low_res - kp_high_res| and |dense_motion_low_res - dense_motion_high_res|
2. Why you need to guarantee this?
